### DISCUSSION TITLE
Auto zoom (Navigation and Follow location)

### DISCUSSION THREAD

**Discussion by Maverikk65** [devemux86/cruiser#680] [Q&A] [answered]
Helloooo... I am working my way into this navigation. In the morning I just drive the 4 km to the office, but in the evening I take a little detour to get accustomized. Firstly I switched the Map theme to Dark already in the office before setting off. That works quite well. The dark map also has an extremely well layout regarding the display of autobahns, main roads and secondary roads and the town names, depending on the zoom factor. Very well, never seen sooo gooood! ;-)

While driving a short route of 3 shaping points and 1 waypoint I was extremely astonished --- the 3 waypoints were not exactly on the road but just a bit off. No problem, I thought --- oh no, I had to delete them manually (while driving, a bit distractiing... :-( ) At home I found (quite by chance) the option BRouter, Advanced, Correct misplaced waypoints (off-100m). I switched to "100m" now and hope it will be better tomorrow... 

However, I have one question for Auto zoom: There are 2 options, Navigation and Follow location. I cannot really imagine what is what and why it is customizable...? Both terms show on other places in the app as well... Pls help/input. 
Thanks, danke, VG, NG.

**Comment by r11d11** [comment]
Hi Maverikk65, you could also try the following options in Cruiser, if you do not want to manualy delete waypoints /Shaping points
Settings
  Navigation
    Strict route following   off

You can enter the "The follow location mode, when you long press on location button at bottom-right.
See also https://github.com/devemux86/cruiser/discussions/611

So I think with Navigation/Auto zoom / Follow location you can turn on or off the auto Zoom in this follow location mode

**Comment by devemux86** [comment, marked as answer]
@Maverikk65 

> the 3 waypoints were not exactly on the road but just a bit off

You can use the crosshair to place the waypoints exactly where you want them.
If waypoints are not located on the road, routers attach them to the nearest road during calculation.

> I have one question for Auto zoom: There are 2 options, Navigation and Follow location.

Auto zoom can be enabled in both modes: navigation and follow location.

Follow mode just follows the GPS location without turn instructions (useful for tracks)
and can be enabled with long press at location button.

**Reply by 0709wiwiwi** [reply]
Shaping and via points should [indeed ](https://github.com/devemux86/cruiser/discussions/680#discussioncomment-11469998) be placed as accurately as possible. 
Shaping and via points are “snapped” by the router to the nearest suitable path.
If snapped to the desired path there is no problem. If snapped to a side road you don't want, the router will anyway send you there and return after only a few meters. Very annoying. You avoid something like this by working accurately and not placing shaping points at intersections but a little further down the road to be followed. 

Cruiser tools. 
**- Place waypoints on street.**
Shaping and via points are visually correctly positioned ON the road.
**- Correct misplaced via points.** (_Only @ normal routing_)
Short in and out track tails will not be generated within the set distance. 
By adding an extra shaping point you can still enforce a track tail.

**Reply by Maverikk65** [reply]
Aah, okay, many thanks. Normally I only use the "Navigation" mode, so I will enable the Auto zoom on both... Maybe not --- when you arrive at destination but have to have an overview about the next convenient parking spot (in a city) you might not want to zoom but to stay in the actual overview... 

On little side question (to Emux): Is it possible to have a lower zoom, so you can see more of the surrounding area or track? I think this is something I would desire... ;-) 

I will keep on testing, exploring, driving... Wrooooommmmmm... 
VG, NG.

**Reply by devemux86** [reply]
@Maverikk65 

> Is it possible to have a lower zoom, so you can see more of the surrounding area or track?

You can select the "Minimum auto zoom" in "Settings | Navigation | Auto zoom".

First enable the "Settings | Application | Extended settings".

<img src="https://github.com/user-attachments/assets/96d1943f-59b6-4cec-bd83-1e23efcd9e3c" width="50%"/><img src="https://github.com/user-attachments/assets/6dcb6278-5e31-4c93-94ae-4d5becbfa8fd" width="50%"/>

**Reply by Maverikk65** [reply]
Hello Emux @devemux86 , yes, many thanks. I mean --- is it possible to have a lower value for Minimum auto zoom like 10 or 9 or 8 or so? Is it just changing one value or is it reprogramming the app? ;-)  

As I use a smartphone in one car and a tablet in the other I already discovered that the same settings would not work on both... (I ran both devices in the one car today in the morning...). 
Danke, VG, NG.

**Reply by devemux86** [reply]
@Maverikk65 So, it's about of a lower min auto zoom than 12?
We can do that, I need to implement it in the app.

**Reply by Maverikk65** [reply]
Hello Emux @devemux86 , this sounds wonderful and like "you have a wish..." ;-) IMHO I would say like "7". Maybe a bit low but users can put in their own desired values... I like to have a big view, overview while driving. I know that other users like to use the navigation like a "terrain following radar" --- I am different in this respect, I concentrate my view on the road and the next curve...

BTW. Is there an option to "reset to basic settings"? I plan to install on a reserve smartphone to make some screenshots for the wiki.

**Reply by devemux86** [reply]
@Maverikk65 Let's start with min zoom 8, where the map starts to show more detail
and we'll see in practice if we need less.

> Is there an option to "reset to basic settings"?

You can create new profiles and reset profiles using the menu on each profile:

<img src="https://github.com/user-attachments/assets/2ebda761-f5fd-400e-ae7f-1b0d105692eb" width="50%"/><img src="https://github.com/user-attachments/assets/13c37e4a-dc2d-4968-812b-3b0709042f7f" width="50%"/>

**Reply by devemux86** [reply]
@Maverikk65 

> I plan to install on a reserve smartphone to make some screenshots for the wiki.

- For screenshots it is better to use the latest https://github.com/devemux86/cruiser/discussions/671

**Reply by Maverikk65** [reply]
So many options... Wow.

**Comment by devemux86** [comment]
Minimum auto zoom can be set to zoom level 8 in **[Cruiser 5.1.0-b14](https://github.com/devemux86/cruiser/discussions/671)**.

**Reply by Maverikk65** [reply]
Wowo, many thanks. 

I really must say I have to get used to "short press" and "long press". I have been using now navigation devices since 15 years or so, but this is a new feature for me. Very good indeed --- if you get used to it ;-)

**Reply by devemux86** [reply]
@Maverikk65 

> "short press" and "long press"

It is common on mobile devices, they understand 2 touch gestures: tap and long press.
Tap is usually for actions and long press for menus.

**Comment by bios4** [comment]
>> Is there an option to "reset to basic settings"?

> You can create new profiles and reset profiles using the menu on each profile

I also already thought that the possibility to reset SINGLE settings (WITHOUT the need to create a whole new profile, just to have ONE setting reset to its defaults) would be a very nice addition.

Any chance this is implemented @devemux86?

**Reply by devemux86** [reply]
@bios4 Reset each setting individually? This is complicated.

**Reply by bios4** [reply]
What a pity... 😞

I ran into situations more than once where I played around with different settings - which then did not turn out how I expected.
Then I wanted to have single changes reset to start over again.

Having to create a whole new profile for that, and apply all other settings to that new profile first, is not very targeted.

But okay, that's how it is...

**Reply by devemux86** [reply]
@bios4 We will see in the future based on the work on other features.

It's easy to see the default of a setting, just check a reset profile or a profile that was never used.

**Reply by WolfgangZ42** [reply]
> Then I wanted to have single changes reset to start over again.

What about to **export** the specific profile before changing it and to re-**import** if the change(s) didn't work.

**Reply by devemux86** [reply]
@WolfgangZ42 @bios4 

Or simply create a new "copy" profile based on our profile to make tests.

**Reply by bios4** [reply]
Sure this is all POSSIBLE - but I'm sure you agree that it would be more CONVENIENT if there were the possibility to reset a single option by one single click... 😉

Anyway: no need to discuss it further, I am perfectly fine that this is not available (yet).

**Reply by Maverikk65** [reply]
Hmmm --- I don't want to "Klugscheißern" --- but it is good practice when testing something that you always change only 1 parameter and look for the result. If you don't get the desired result you switch back. Because --- if you change multiple parameters at once you might end up with undesired results... /Klugscheissermodus aus...
VG, NG.
